runbooks/how-to-do-a-deployment.md

```diff
@@ -0,0 +1,120 @@
+
+1. Ensure that no other jobs are currently building or queued up to build; in addition, one should check if another developer is performing work/testing in the `Test` environment.
+To do so, one can attempt to notify other developers that you will be initiating a build by posting a message to the `Slack` channel `#bfd-builds`:
+    - post message: `/dibs on test` to stake your claim on the test enviornemnt.
+
+![Dibs-On](resources/deploy-dibs-on.png)    
+If you are not able to stake your claim to the `Test` environment, it would mean that another developer has it reserved for their build and/or testing. In that scenario one would probably not proceed unless `Test` is available; the prudent move here is to reach out to the developer to see if they are able to relinquish their reservation of the `Test` environment.
+
+2. From browser navigate to CloudBees Managed Controller: https://jenkins-east.cloud.cms.gov/bfd/
+
+3. The `Managed Controller` page for BFD provides a list of build configurations available in the CloudBees Jenkins server. 
+
+`OPTIONAL`:
+Prior to moving on to next step(s) associated with the BFD Pipeline, it is recommended that you consider locking target environment(s) to preclude an unintended deployment or another build from over-writing your planned deployment.
+    - On the left-side list of operations, near the bottom is an item `Lockable Resources`.
+    ![Lockable](resources/deploy-lockable.png)
+    - selecting the `Lockable Resources` brings up a small dailog box that allows you to lock one or more of the BFD environments.
+     ![Lockable Resources](resources/deploy-lockable-resources.png)
+    - use the `env_test` `Reserve` button to create a exclusive deployment lock on the `Test` environment.
+    ![Lockable](resources/deploy-reserve-test.png) 
+
+This notion of locking an environment will likely be going away as determined by future work in the Jenkins enviroment. 
+
+4. Select a Jenkins job configuration to be used for the deployment; for a standard BFD build/deployment, select the `BFD - Multibranch and Multistage Pipeline` link.
+
+     ![Lockable Resources](resources/deploy-multi-branch.png)
+
+The name `Multistage Pipeline` suggests a series of tasks will be executed; of primary interest are the target environments for a deployment:
+- build and deploy to the `test` environment.
+- build and deploy to the `prod-sbx` environment.
+- build and deploy to the `prod` environment.
+
+There are some optional pipeline tasks such as a doing an AMI build, that can be included in the pipeline stages based on operator specified parameters. In addition there are pipeline fail-safes that require operator intervention to proceed to the next stage; for example, to proceed from `test` to `prod-sbx` build/deploy, the operator may be asked to respond to a `Proceed` dialog button.
```

Comment:
Would be helpful to describe the AMI in brief and when we actually want to build it that as part of the deploy.

Reply:
Even a small aside with that information is useful; reading this from ignorance, I wouldn't know that the AMI build timing is arbitrary/adhoc.